File: app/src/main/java/de/kontranik/freebudget/adapter/DrawerItemCustomAdapter.java
package de.kontranik.freebudget.adapter;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import de.kontranik.freebudget.R;
import de.kontranik.freebudget.model.DrawerItem;

public class DrawerItemCustomAdapter extends ArrayAdapter<DrawerItem> {
    Context mContext;
    int layoutResourceId;
    DrawerItem data[] = null;

    public DrawerItemCustomAdapter(Context mContext, int layoutResourceId, DrawerItem[] data) {

        super(mContext, layoutResourceId, data);
        this.layoutResourceId = layoutResourceId;
        this.mContext = mContext;
        this.data = data;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        View listItem = convertView;

        LayoutInflater inflater = ((Activity) mContext).getLayoutInflater();
        listItem = inflater.inflate(layoutResourceId, parent, false);

        ImageView imageViewIcon = (ImageView) listItem.findViewById(R.id.imageViewIcon);
        TextView textViewName = (TextView) listItem.findViewById(R.id.textViewName);

        DrawerItem folder = data[position];


        imageViewIcon.setImageResource(folder.icon);
        textViewName.setText(folder.name);

        return listItem;
    }
}

File: app/src/main/java/de/kontranik/freebudget/adapter/DrawerItemCustomAdapter.kt
package de.kontranik.freebudget.adapter

import de.kontranik.freebudget.model.DrawerItem
import android.widget.ArrayAdapter
import android.view.ViewGroup
import android.app.Activity
import android.content.Context
import android.view.View
import android.widget.ImageView
import de.kontranik.freebudget.R
import android.widget.TextView

class DrawerItemCustomAdapter(
    var mContext: Context,
    var layoutResourceId: Int,
    var data: Array<DrawerItem>
) : ArrayAdapter<DrawerItem>(
    mContext, layoutResourceId, data
) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var listItem = convertView
        val inflater = (mContext as Activity).layoutInflater
        listItem = inflater.inflate(layoutResourceId, parent, false)
        val imageViewIcon = listItem.findViewById<View>(R.id.imageViewIcon) as ImageView
        val textViewName = listItem.findViewById<View>(R.id.textViewName) as TextView
        val folder = data[position]
        imageViewIcon.setImageResource(folder.icon)
        textViewName.text = folder.name
        return listItem
    }

}